src/screens/earn/farm-item/farm-item.tsx

```diff
@@ -0,0 +1,138 @@
+import { Bage } from 'src/components/bage/bage';
+import { ButtonLargePrimary } from 'src/components/button/button-large/button-large-primary/button-large-primary';
+import { ButtonLargeSecondary } from 'src/components/button/button-large/button-large-secondary/button-large-secondary';
+import { Divider } from 'src/components/divider/divider';
+import { IconNameEnum } from 'src/components/icon/icon-name.enum';
+import { useReadOnlyTezosToolkit } from 'src/hooks/use-read-only-tezos-toolkit.hook';
+import { ConfirmationTypeEnum } from 'src/interfaces/confirm-payload/confirmation-type.enum';
+import { FarmToken } from 'src/interfaces/earn.interface';
+import { ModalsEnum } from 'src/navigator/enums/modals.enum';
+import { useNavigation } from 'src/navigator/hooks/use-navigation.hook';
+import { UserStakeValueInterface } from 'src/store/farms/state';
+import { navigateAction } from 'src/store/root-state.actions';
+import { useSelectedAccountSelector } from 'src/store/wallet/wallet-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { TEZ_TOKEN_SLUG } from 'src/token/data/tokens-metadata';
+import { isDefined } from 'src/utils/is-defined';
+import { mutezToTz } from 'src/utils/tezos.util';
+
+import { FarmTokens } from '../farm-tokens/farm-tokens';
+import { useFarmItemStyles } from './farm-item.styles';
+
+interface Props {
+  farm: SingleFarmResponse;
+  lastStakeRecord: UserStakeValueInterface;
+}
+
+const FARM_PRECISION = 18;
+const DEFAULT_AMOUNT = 0;
+const DEFAULT_EXHANGE_RATE = 1;
+const DEFAULT_DECIMALS = 2;
+const REWARDS_DECIMALS = 6;
+
+export const FarmItem: FC<Props> = ({ farm, lastStakeRecord }) => {
+  const styles = useFarmItemStyles();
+  const dispatch = useDispatch();
+  const { navigate } = useNavigation();
+  const selectedAccount = useSelectedAccountSelector();
+  const tezos = useReadOnlyTezosToolkit(selectedAccount);
+
+  const rewardToken: FarmToken = {
+    symbol: farm.item.rewardToken.metadata.symbol,
+    thumbnailUri: farm.item.rewardToken.metadata.thumbnailUri
+  };
+
+  const stakeTokens = farm.item.tokens.map(token => {
+    const result: FarmToken = {
+      symbol: token.metadata.symbol,
+      thumbnailUri: token.metadata.thumbnailUri
+    };
+
+    if (token.metadata.symbol.toLowerCase() === TEZ_TOKEN_SLUG) {
+      result.iconName = IconNameEnum.TezToken;
+    }
+
+    return result;
+  });
+
+  const apr = isDefined(farm.item.apr) ? new BigNumber(farm.item.apr).toFixed(DEFAULT_DECIMALS) : '---';
+  const depositAmountAtomic = mutezToTz(
+    new BigNumber(lastStakeRecord?.depositAmountAtomic ?? DEFAULT_AMOUNT),
+    FARM_PRECISION
+  ).multipliedBy(farm.item.depositExchangeRate ?? DEFAULT_EXHANGE_RATE);
+  const claimableRewardsAtomic = mutezToTz(
+    new BigNumber(lastStakeRecord?.claimableRewards ?? DEFAULT_AMOUNT),
+    farm.item.rewardToken.metadata.decimals
+  ).multipliedBy(farm.item.earnExchangeRate ?? DEFAULT_EXHANGE_RATE);
```

Comment:
Use `useMemo` for all of these variables to prevent memory over usage because of creating objects instances on each render